diagnostic_menu.py

```diff
@@ -0,0 +1,207 @@
+
+
+        layout = QVBoxLayout()
+        layout.setAlignment(Qt.AlignTop)
+        layout.addWidget(self.logs_label)
+
+        widget = QWidget()
+        widget.setLayout(layout)
+        self.timer = QTimer(self)
+
+        self.timer.timeout.connect(lambda: self.updateLogLabel(f'{ASSETS_FOLDER}/logs/{date.today()}.log'))
+        self.timer.start(1000)
+        
+
+        self.setWidget(widget)
+        self.setWidgetResizable(True)
+
+
+
+    def openLogs(self, file):
+        try:
+            with open(file, 'r') as txt:
+                list_of_lines = txt.read() 
+            self.logs = list_of_lines
+        except FileNotFoundError:
+            logger.error(f'File {file} not found')
+            self.logs = 'File not found'
+        
+    
+    def updateLogLabel(self, file):
+        with open(file, 'r') as txt:
+            list_of_lines = txt.read() 
+        self.logs_label.setText(list_of_lines)
+
+    def keyPressEvent(self, qKeyEvent):
+        if qKeyEvent.key() == Qt.Key_Escape: 
+            self.close()
+        else:
+            super().keyPressEvent(qKeyEvent)
+
+class Menu(QMainWindow):
+    sig = pyqtSignal()
+
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle('Menu')
+        self.setGeometry(400, 400, 700, 500)
+        self.logs_window = None
+        self.json_window = None
+
+
+
+        default_font = QFont()
+        default_font.setBold(True)
+        layout = QVBoxLayout()
+
+        
+        self.logs_btn = QPushButton(text='Open Logs')
+        self.logs_btn.clicked.connect(self.openLogs)        
+        self.json_btn = QPushButton(text='Open Company Data')
+        self.json_btn.clicked.connect(self.openJson)
+        storage_info = self.getStorageInfo()
+        self.storage_label = QLabel(text=f'Storage: \nTotal: {storage_info["total"]}GB | Used: {storage_info["used"]}GB | Free: {storage_info["free"]}GB')
+        self.storage_label.setStyleSheet("font-size: 15px;")
+        self.storage_label.setAlignment(Qt.AlignCenter)
+        self.storage_label.setFont(default_font)
+
+        self.cpu_util_label = QLabel(text=f'{self.getCpuInfo()}')
+        self.cpu_util_label.setAlignment(Qt.AlignCenter)
+        self.cpu_util_label.setFont(default_font)
+        self.cpu_util_label.setStyleSheet("font-size: 15px;")
+
+        self.info_label = QLabel(text='(To exit press ESC)')
+        self.info_label.setAlignment(Qt.AlignCenter)
+        self.info_label.setFont(default_font)
+
+        layout.addWidget(self.info_label)
+        layout.addWidget(self.json_btn)
+        layout.addWidget(self.logs_btn)
+        layout.addWidget(self.storage_label)
+        layout.addWidget(self.cpu_util_label)
+
+        widget = QWidget()
+        widget.setLayout(layout)
+        self.setCentralWidget(widget)
+        self.timer = QTimer(self)
+
+        self.timer.timeout.connect(self.updateMenu)
+        self.timer.start(1000)
+        
+
+    def openLogs(self):
```

Comment:
According to your code below, it doesn't open, but toggles instead.

Reply:
✅